flytekit/annotated/context_manager.py

```diff
@@ -243,6 +254,36 @@ def new_compilation_context(self) -> Generator['FlyteContext', None, None]:
         finally:
             FlyteContext.OBJS.pop()
 
+    def enter_conditional_section(self):
```

Comment:
Since these aren't real contexts, i feel like it would make more sense to put them into ExecutionState, so as to keep the top level context manager cleaner.

Reply:
yup